Tools/autotest/quadplane.py

```diff
@@ -78,6 +78,147 @@ def get_disarm_delay(self):
+        self.set_parameter("RC%d_OPTION" % arm_ch, armdisarm_option)
+
+        """arm with GCS, record Motor1 SPIN_ARM PWM output and disarm"""
+        spool_delay = self.get_parameter("Q_M_SPOOL_TIME") + 0.25
+        self.zero_throttle()
+        self.arm_vehicle()
+        self.progress("Waiting for Motor1 to spool up to SPIN_ARM")
+        self.delay_sim_time(spool_delay)
+        spin_arm_pwm = self.wait_servo_channel_value(5, min_pwm, comparator=operator.gt)
+        self.progress("spin_arm_pwm: %d" % spin_arm_pwm)
+        self.disarm_vehicle()
+
+        """arm with switch, record Motor1 SPIN_MIN PWM output and disarm"""
+        self.set_rc(8, 2000)
+        self.delay_sim_time(spool_delay)
+        self.progress("Waiting for Motor1 to spool up to SPIN_MIN")
+        spin_min_pwm = self.wait_servo_channel_value(5, spin_arm_pwm, comparator=operator.gt)
+        self.progress("spin_min_pwm: %d" % spin_min_pwm)
+        self.set_rc(8, 1000)
+
+        if (spin_arm_pwm >= spin_min_pwm):
+            raise PreconditionFailedException("SPIN_MIN pwm not greater than SPIN_ARM pwm")
+
+        """verify that arming with switch results in higher PWM output"""
+        self.progress("########## Test auxswitch arming with Q_OPTIONS=AirMode")
+        modes = ('QSTABILIZE', 'QACRO')
+        for mode in modes:
+            self.progress("########## Testing %s mode" % mode)
+            self.change_mode(mode)
+            self.zero_throttle()
+            self.progress("arming with switch at zero throttle")
+            self.arm_motors_with_switch(arm_ch)
+            self.progress("Waiting for Motor1 to speed up")
+            self.wait_servo_channel_value(5, spin_min_pwm, comparator=operator.ge)
+
+            self.progress("disarming with switch")
+            self.disarm_motors_with_switch(arm_ch)
+            self.progress("Waiting for Motor1 to stop")
+            self.wait_servo_channel_value(5, min_pwm, comparator=operator.le)
+            self.wait_ready_to_arm()
+
+        """verify that arming with switch does not result in higher PWM output"""
+        modes = ('QLOITER', 'QHOVER')
+        for mode in modes:
+            self.progress("########## Testing %s mode" % mode)
+            self.change_mode(mode)
+            self.zero_throttle()
+            self.progress("arming with switch at zero throttle")
+            self.arm_motors_with_switch(arm_ch)
+            self.progress("Waiting for Motor1 to (not) speed up")
+            self.delay_sim_time(spool_delay)
+            self.wait_servo_channel_value(5, spin_arm_pwm, comparator=operator.le)
+
+            self.progress("disarming with switch")
+            self.disarm_motors_with_switch(arm_ch)
+            self.progress("Waiting for Motor1 to stop")
+            self.wait_servo_channel_value(5, min_pwm, comparator=operator.le)
+            self.wait_ready_to_arm()
+
+        """set  RC7_OPTION to AIRMODE"""
+        """verify that RC7 turns airmode on/off while armed"""
```

Comment:
This is somewhere `start_subtest` might be good.
```suggestion
        self.start_subtest("verify that RC7 turns airmode on/off while armed")
```

Reply:
done